### ISSUE TITLE
Overwriting the stack trace makes debugging hard

### ISSUE THREAD

**Issue by j-baker** [palantir/conjure-java-runtime#25]
The error handler currently replaces the system stack trace with the stack trace from the server. This can be quite frustrating. For example, I've received IllegalArgumentExceptions after doing a refactor and known that it could have come from one of a few places in my code (and have no immediate way of finding out which), but know for sure that it came from a part of the server that throws IllegalArgumentExceptions.

I haven't thought of a great solution yet (perhaps errors with status <500 could not have their stack trace propagated?), but I wanted to flag this (happy to implement if this is seen as reasonable).

Below is an example of this being the case, accidentally leaving "Bearer" off an auth header. I regard this error trace as being useless.

```
java.lang.IllegalArgumentException: Unexpected HTTP Authorization header format "eyJhbGciOiJSUzUxMiJ9.eyJzdWIiOiIzNjQzODJhYy1jNmY4LTQ2Y2QtYmRiZi05ZWM0Mjc1NzgzMTUiLCJzdWJqZWN0TmFtZSI6ImFkbWluIiwidG9rZW5OYW1lIjoidG9rZW41IiwianRpIjoiYzZmODhhMmItMmQxYi00NzQwLWJiYzItZDYyMTcyYjkyNDhkIn0.qVheamwOwd_Ucl-F7zp2F30uAyAFMVy6sDCPMslMywk3DZHOdlBXuYn5hqehWpg1vnUH7rGWzoxxrpe2AMbPyHYWfP96IGOOZBdAsxN5psz6f-GkLANaShOWqtdn2YDz80PQQBgiP6zmASw6WvenwAaeECSrgtZYrNXuApr51Dhp5S5T3neq02R8b-tUhDH9_NObu8V2UTXM5pORaMckW0LW1fgkM3Z843dUvo4KWDiKbJ6vEj2z8KUKNs5PWowoDNi94XJi2Bz3YNQRrVITkGxfG24bqTxggoFlmZSeHkaAjkKDuv2-HMnX_FD9b0mEvNASqmLK8UAAaKXHSkP4_0rw0rG9mdRUxpSZnAwhYJvE4NDUFmwtoLq9nSMYmLXq0uZ39w0fTLR28qZdx6kjN699GE9nZZptUpJaMkram64KktbZ7ylWVm6dMy_HJWR7gIiroreDH3NtVFDiPKmb5d0crgApmvtzKbOJeGH08xDlSOf7Zl3A9S8HP0oPPUKIv4TdDxjf5f7js7iWBdjBF8lKCQ2W1HoPPeogxH1VVUOmyumCTt6nxxmUNYpIUK-ysExGn1ym_xaGQSp0atcrHznGeeGRFh9l-dGCwowzb9WiXOBDxwCer-LIiFt9zkHZUDo9CIHyA5PHPXx7rW-4gFuOucN52nogmoVhV7nTvV4"
    at com.palantir.AppName.auth.BearerAuthService.authenticateHeader(BearerAuthService.java:59)
    at com.palantir.AppName.server.AppName.registerResource(AppName.java:120)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:497)
    at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
    at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205)
    at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347)
    at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102)
    at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:308)
    at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271)
    at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:315)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:297)
    at org.glassfish.jersey.internal.Errors.process(Errors.java:267)
    at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317)
    at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:291)
    at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1140)
    at org.glassfish.jersey.servlet.WebComponent.service(WebComponent.java:403)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:386)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:334)
    at org.glassfish.jersey.servlet.ServletContainer.service(ServletContainer.java:221)
    at io.dropwizard.jetty.NonblockingServletHolder.handle(NonblockingServletHolder.java:49)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1669)
    at org.eclipse.jetty.servlets.UserAgentFilter.doFilter(UserAgentFilter.java:83)
    at org.eclipse.jetty.servlets.GzipFilter.doFilter(GzipFilter.java:300)
    at io.dropwizard.jetty.BiDiGzipFilter.doFilter(BiDiGzipFilter.java:134)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
    at io.dropwizard.servlets.ThreadNameFilter.doFilter(ThreadNameFilter.java:29)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
    at io.dropwizard.jersey.filter.AllowedMethodsFilter.handle(AllowedMethodsFilter.java:44)
    at io.dropwizard.jersey.filter.AllowedMethodsFilter.doFilter(AllowedMethodsFilter.java:39)
    at org.eclipse.jetty.servlet.ServletHandler$CachedChain.doFilter(ServletHandler.java:1652)
    at org.eclipse.jetty.servlet.ServletHandler.doHandle(ServletHandler.java:585)
    at org.eclipse.jetty.server.handler.ContextHandler.doHandle(ContextHandler.java:1127)
    at org.eclipse.jetty.servlet.ServletHandler.doScope(ServletHandler.java:515)
    at org.eclipse.jetty.server.handler.ContextHandler.doScope(ContextHandler.java:1061)
    at org.eclipse.jetty.server.handler.ScopedHandler.handle(ScopedHandler.java:141)
    at org.eclipse.jetty.server.handler.HandlerWrapper.handle(HandlerWrapper.java:97)
```

**Comment by markelliot** [comment]
I see, so the request is basically that we include the client stack trace on top of this, should be possible.

**Comment by j-baker** [comment]
yep - i considered appending, but Jersey exceptions can be huge (that is truncated, I think), and two of them concatenated might be unpleasant.

**Comment by markelliot** [comment]
Current thinking is we can instantiate the exception class twice and nest the server exception inside a client exception so you get both stack traces and that java printing of stack traces will shorten as appropriate. (happy to take a PR! :))

**Comment by j-baker** [comment]
Ah, ok. Didn't know you could do that (although in hindsight it's obvious). Yep, I can do that.

**Comment by markelliot** [comment]
this was fixed in 0.5.0.
